--- old/core/modules/user/src/Controller/UserAuthenticationController.php	2020-05-29 09:22:19.780000000 -0400
+++ web/core/modules/user/src/Controller/UserAuthenticationController.php	2020-05-29 09:50:51.760000000 -0400
@@ -404,4 +404,23 @@
     return '';
   }
 
+      /**
+   * Gets user logout token
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The request.
+   *
+   * @return \Symfony\Component\HttpFoundation\Response
+   *   A response which contains the ID and CSRF token.
+   */
+  public function logoutToken(Request $request) {
+    $format = $this->getRequestFormat($request);
+      $response_data = [];
+      $logout_route = $this->routeProvider->getRouteByName('user.logout.http');
+      // Trim '/' off path to match \Drupal\Core\Access\CsrfAccessCheck.
+      $logout_path = ltrim($logout_route->getPath(), '/');
+      $response_data['logout_token'] = $this->csrfToken->get($logout_path);
+      $encoded_response_data = $this->serializer->encode($response_data, $format);
+      return new Response($encoded_response_data);
+  }
 }
--- old/core/modules/user/user.routing.yml	2020-05-29 09:22:19.790000000 -0400
+++ web/core/modules/user/user.routing.yml	2020-05-29 09:48:55.930000000 -0400
@@ -210,6 +210,16 @@
     _maintenance_access: TRUE
     no_cache: TRUE
 
+user.logout.token:
+  path: '/session/logouttoken'
+  defaults:
+    _controller: '\Drupal\user\Controller\UserAuthenticationController::logoutToken'
+    _title: 'Logout Token'
+  methods: [GET]
+  requirements:
+    _user_is_logged_in: 'TRUE'
+    _format: 'json'
+
 user.well-known.change_password:
   path: '/.well-known/change-password'
   defaults:
